From 1154ec919b912e28d5f75c0dc8cdaf0b13a9abc8 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Tue, 21 Apr 2015 09:28:34 -0400
Subject: [PATCH] support system wise install ijulia

---
 deps/build.jl | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/deps/build.jl b/deps/build.jl
index d723ff2..5cd2033 100644
--- a/deps/build.jl
+++ b/deps/build.jl
@@ -81,7 +81,8 @@ if VERSION >= v"0.3"
     push!(kernelcmd_array,"-i")
 end
 
-append!(kernelcmd_array, ["-F", escape_string(joinpath(Pkg.dir("IJulia"),"src","kernel.jl")), "{connection_file}"])
+ijulia_dir = get(ENV, "IJULIA_DIR", Pkg.dir("IJulia"))
+append!(kernelcmd_array, ["-F", escape_string(joinpath(ijulia_dir,"src","kernel.jl")), "{connection_file}"])
 
 
 
-- 
2.3.5

