diff --git a/cmake/FindPhononInternal.cmake b/cmake/FindPhononInternal.cmake
index ca884ec..fefe2d4 100644
--- a/cmake/FindPhononInternal.cmake
+++ b/cmake/FindPhononInternal.cmake
@@ -71,7 +71,7 @@ find_package(Qt4 ${_REQ_STRING_KDE4})
 # - Automoc
 
 # Starting with CMake 2.8.6 there is a builtin to replace automoc4, use that when possible.
-if(CMAKE_VERSION VERSION_GREATER 2.8.5)
+if(Off)
     message(STATUS "Using CMake automoc builtin")
     set(CMAKE_AUTOMOC TRUE)
     # Compatiblity Macros
@@ -81,7 +81,7 @@ if(CMAKE_VERSION VERSION_GREATER 2.8.5)
     macro(AUTOMOC4_ADD_LIBRARY _target_NAME _add_executable_param)
         add_library(${_target_NAME} ${_add_executable_param} ${ARGN})
     endmacro(AUTOMOC4_ADD_LIBRARY)
-else(CMAKE_VERSION VERSION_GREATER 2.8.5)
+else()
     message(STATUS "Can not use CMake automoc builtin, trying to find automoc4")
     find_package(Automoc4 REQUIRED)
     if (NOT AUTOMOC4_VERSION)
@@ -91,7 +91,7 @@ else(CMAKE_VERSION VERSION_GREATER 2.8.5)
     if (NOT _automoc4_version_ok)
         message(FATAL_ERROR "Your version of automoc4 is too old. You have ${AUTOMOC4_VERSION}, you need at least 0.9.86")
     endif (NOT _automoc4_version_ok)
-endif(CMAKE_VERSION VERSION_GREATER 2.8.5)
+endif()
 
 # restore the original CMAKE_MODULE_PATH
 set(CMAKE_MODULE_PATH ${_phonon_cmake_module_path_back})
