diff -ur libappindicator-12.10.0.orig/bindings/mono/libappindicator-api.metadata libappindicator-12.10.0/bindings/mono/libappindicator-api.metadata
--- libappindicator-12.10.0.orig/bindings/mono/libappindicator-api.metadata	2013-02-14 10:16:28.074779005 -0500
+++ libappindicator-12.10.0/bindings/mono/libappindicator-api.metadata	2013-02-14 10:59:16.914690745 -0500
@@ -13,6 +13,8 @@
 	<attr path="/api/namespace/object[@cname='AppIndicator']/signal[@field_name='connection_changed']" name="cname">connection-changed</attr>
 	<attr path="/api/namespace/object[@cname='AppIndicator']/signal[@field_name='scroll_event']" name="name">ScrollEvent</attr>
 	<attr path="/api/namespace/object[@cname='AppIndicator']/signal[@field_name='scroll_event']" name="cname">scroll-event</attr>
+	<attr path="/api/namespace/object[@cname='AppIndicator']/signal[@field_name='activate_event']" name="name">ActivateEvent</attr>
+	<attr path="/api/namespace/object[@cname='AppIndicator']/signal[@field_name='activate_event']" name="cname">activate-event</attr>
 	<attr path="/api/namespace/object[@cname='AppIndicator']/signal[@field_name='new_icon']" name="name">NewIcon</attr>
 	<attr path="/api/namespace/object[@cname='AppIndicator']/signal[@field_name='new_icon']" name="cname">new-icon</attr>
 	<attr path="/api/namespace/object[@cname='AppIndicator']/signal[@field_name='new_icon_theme_path']" name="name">NewIconThemePath</attr>
diff -ur libappindicator-12.10.0.orig/bindings/mono/libappindicator-api.raw libappindicator-12.10.0/bindings/mono/libappindicator-api.raw
--- libappindicator-12.10.0.orig/bindings/mono/libappindicator-api.raw	2013-02-14 10:16:28.074779005 -0500
+++ libappindicator-12.10.0/bindings/mono/libappindicator-api.raw	2013-02-14 10:59:16.914690745 -0500
@@ -91,6 +91,13 @@
           <parameter type="gpointer" name="user_data" />
         </parameters>
       </signal>
+      <signal when="LAST" field_name="activate_event">
+        <return-type type="void" />
+        <parameters>
+          <parameter type="AppIndicator*" name="indicator" />
+          <parameter type="gpointer" name="user_data" />
+        </parameters>
+      </signal>
       <virtual_method name="Fallback" cname="fallback">
         <return-type type="GtkStatusIcon*" />
         <parameters>
@@ -242,4 +249,4 @@
       </method>
     </object>
   </namespace>
-</api>
\ No newline at end of file
+</api>
diff -ur libappindicator-12.10.0.orig/src/app-indicator.c libappindicator-12.10.0/src/app-indicator.c
--- libappindicator-12.10.0.orig/src/app-indicator.c	2013-02-14 10:16:28.074779005 -0500
+++ libappindicator-12.10.0/src/app-indicator.c	2013-02-14 11:00:08.567859604 -0500
@@ -107,6 +107,7 @@
 	CONNECTION_CHANGED,
 	NEW_ICON_THEME_PATH,
 	SCROLL_EVENT,
+        ACTIVATE_EVENT,
 	LAST_SIGNAL
 };
 
@@ -538,6 +539,21 @@
 	                                  _application_service_marshal_VOID__INT_UINT,
 	                                  G_TYPE_NONE, 2, G_TYPE_INT, G_TYPE_UINT);
 
+	/**
+	 * AppIndicator::activate-event:
+	 * @arg0: The #AppIndicator object
+	 *
+	 * Signaled when the #AppIndicator receives a scroll event.
+	 */
+	signals[ACTIVATE_EVENT] =
+            g_signal_new(APP_INDICATOR_SIGNAL_ACTIVATE_EVENT,
+                         G_TYPE_FROM_CLASS(klass),
+                         G_SIGNAL_RUN_LAST,
+                         G_STRUCT_OFFSET (AppIndicatorClass, activate_event),
+                         NULL, NULL,
+                         NULL,
+                         G_TYPE_NONE, 0);
+
 	/* DBus interfaces */
 	if (item_node_info == NULL) {
 		GError * error = NULL;
@@ -1092,6 +1108,8 @@
 		{
 			gtk_widget_activate (menuitem);
 		}
+	} else if (g_strcmp0(method, "Activate") == 0) {
+            g_signal_emit(G_OBJECT(app), signals[ACTIVATE_EVENT], 0);
 	} else {
 		g_warning("Calling method '%s' on the app-indicator and it's unknown", method);
 	}
@@ -1660,19 +1678,7 @@
 static void
 status_icon_activate (GtkStatusIcon * icon, gpointer data)
 {
-	GtkMenu * menu = app_indicator_get_menu(APP_INDICATOR(data));
-	if (menu == NULL)
-		return;
-	
-	gtk_menu_popup(menu,
-	               NULL, /* Parent Menu */
-	               NULL, /* Parent item */
-	               gtk_status_icon_position_menu,
-	               icon,
-	               1, /* Button */
-	               gtk_get_current_event_time());
-
-	return;
+    g_signal_emit(G_OBJECT(data), signals[ACTIVATE_EVENT], 0);
 }
 
 /* Handles the right-click action by the status icon by showing
diff -ur libappindicator-12.10.0.orig/src/app-indicator.h libappindicator-12.10.0/src/app-indicator.h
--- libappindicator-12.10.0.orig/src/app-indicator.h	2013-02-14 10:16:28.074779005 -0500
+++ libappindicator-12.10.0/src/app-indicator.h	2013-02-14 11:00:39.544428402 -0500
@@ -119,6 +119,7 @@
 #define APP_INDICATOR_SIGNAL_CONNECTION_CHANGED  "connection-changed"
 #define APP_INDICATOR_SIGNAL_NEW_ICON_THEME_PATH "new-icon-theme-path"
 #define APP_INDICATOR_SIGNAL_SCROLL_EVENT        "scroll-event"
+#define APP_INDICATOR_SIGNAL_ACTIVATE_EVENT      "activate-event"
 
 /**
  * AppIndicatorCategory:
@@ -228,7 +229,7 @@
 	void (*app_indicator_reserved_3)(void);
 	void (*app_indicator_reserved_4)(void);
 	void (*app_indicator_reserved_5)(void);
-	void (*app_indicator_reserved_6)(void);
+	void (*activate_event)(AppIndicator *indicator, gpointer user_data);
 };
 
 /**
diff -ur libappindicator-12.10.0.orig/src/notification-item.xml libappindicator-12.10.0/src/notification-item.xml
--- libappindicator-12.10.0.orig/src/notification-item.xml	2013-02-14 10:16:28.074779005 -0500
+++ libappindicator-12.10.0/src/notification-item.xml	2013-02-14 10:59:16.914690745 -0500
@@ -20,6 +20,10 @@
 		<property name="XAyatanaOrderingIndex" type="u" access="read" />
 
 <!-- Methods -->
+		<method name="Activate">
+			<arg type="i" name="x" direction="in" />
+			<arg type="i" name="y" direction="in" />
+		</method>
 		<method name="Scroll">
 			<arg type="i" name="delta" direction="in" />
 			<arg type="s" name="orientation" direction="in" />
