From 194c1b386752aaff40efc41b84b8db1604ecdbe2 Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Sat, 18 May 2013 22:20:25 -0400
Subject: [PATCH] -

---
 db.txt    | 3 +++
 db2bin.py | 2 +-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/db.txt b/db.txt
index c5861b8..f1654a0 100644
--- a/db.txt
+++ b/db.txt
@@ -797,3 +797,6 @@ country ZA:
 country ZW:
 	(2402 - 2482 @ 40), (N/A, 20)
 
+country HX:
+	(4910 - 5835 @ 40), (N/A, 35)
+	(2402 - 2494 @ 40), (N/A, 35)
diff --git a/db2bin.py b/db2bin.py
index 41d3741..390a431 100755
--- a/db2bin.py
+++ b/db2bin.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 
 from cStringIO import StringIO
 import struct
-- 
1.8.2.3

