diff -crB pidgin.orig/pidgin/gtkconv.c pidgin/pidgin/gtkconv.c
*** pidgin.orig/pidgin/gtkconv.c	2012-06-11 07:06:48.852051370 -0400
--- pidgin/pidgin/gtkconv.c	2012-08-07 08:43:23.004953374 -0400
***************
*** 2269,2277 ****
  		return FALSE;
  	}
  
! 	if (event->type == GDK_KEY_RELEASE)
! 		gtk_widget_grab_focus(gtkconv->entry);
! 
  	gtk_widget_event(gtkconv->entry, (GdkEvent *)event);
  
  	return TRUE;
--- 2269,2275 ----
  		return FALSE;
  	}
  
! 	gtk_widget_grab_focus(gtkconv->entry);
  	gtk_widget_event(gtkconv->entry, (GdkEvent *)event);
  
  	return TRUE;
